extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -64,12 +66,14 @@ export class ProjectsController {
 	private buildHelper: BuildHelper;
 	private buildInfo: DashboardData[] = [];
 	private publishInfo: PublishData[] = [];
+	private deployService: DeployService;
 
 	projFileWatchers = new Map<string, vscode.FileSystemWatcher>();
 
-	constructor() {
-		this.netCoreTool = new NetCoreTool();
+	constructor(private _outputChannel: vscode.OutputChannel) {
+		this.netCoreTool = new NetCoreTool(this._outputChannel);
 		this.buildHelper = new BuildHelper();
+		this.deployService = new DeployService(this._outputChannel);
```

Comment:
It looks like both NetCoreTool and DeployService could have `outputChannel` be undefined since they always check to see if it's set before use (which isn't very often).  If so, then the `projectController` constructor arg could also be nullable, and you wouldn't have to update all the calls to `projectController` in the tests to include a new param.

Reply:
I made them both not nullable

Reply:
Aight, that works too.